corona-warn-app/cwa-app-android
# Risk change dialog pops up after being dismissed (EXPOSUREAPP-3536)

### Description



This PR fixes a bug where the LiveData from the VM wasn't getting updated anymore, causing a dialog to show up every time you traveled to another screen and then returned.
Issue was caused by LiveData being filtered and once it had a value, the filter was ignoring any other state the LiveData could be in.

How to quickly test:

In RiskLevelTransaction, comment out the start() function and replace it with this one:

```kotlin
suspend fun start() = lockAndExecute(
        scope = transactionScope,
        timeout = RISK_LEVEL_TRANSACTION_TIMEOUT
    ) {
        lastCalculatedRiskLevelScoreForRollback.set(RiskLevelRepository.getLastCalculatedScore())
        executeUpdateRiskLevelScore(INCREASED_RISK)
        executeRiskLevelCalculationDateUpdate()
        executeClose()
        return@lockAndExecute
    }
```

- Build & run the app with the "hacked" version of the start function to put the app in a high risk state.
- Change the INCREASED_RISK to LOW_LEVEL_RISK in executeUpdateRiskLevelScore() and build & run again.
- App should display the dialog with the low risk card in the background.
- Pressing Ok on the dialog dismisses the dialog.
- Navigating to a different screen and returning to the home screen won't result in a new dialog being displayed.
